dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/CurrentUserUtil.java

```diff
@@ -177,15 +170,11 @@ public static void setUserSetting(UserSettingKey key, Serializable value) {
 
   private static void setUserSettingInternal(String key, Serializable value) {
     UserDetails currentUser = getCurrentUserDetails();
-    if (currentUser != null) {
-      Map<String, Serializable> userSettings = currentUser.getUserSettings();
-      if (userSettings != null) {
-        if (value != null) {
-          userSettings.put(key, value);
-        } else {
-          userSettings.remove(key);
-        }
-      }
+    Map<String, Serializable> userSettings = currentUser.getUserSettings();
+    if (value != null) {
+      userSettings.put(key, value);
+    } else {
+      userSettings.remove(key);
```

Comment:
I know this is not part of this PR but seeing this seems a little odd.
The method's intention is to set a user setting, but if the value passed is null we remove the key. This sounds like a hidden side effect.